File: 백준/Silver/1138. 한 줄로 서기/한 줄로 서기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        ArrayList<Integer> li = new ArrayList<>();
        ArrayList<Integer> result = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            li.add(Integer.parseInt(st.nextToken()));
            result.add(0);
        }

        for (int i = 0; i < n; i++) {
            int cnt = 0;
            for (int j = 0; j < n; j++) {
                if (cnt == li.get(i) && result.get(j) == 0) {
                    result.set(j,i+1);
                    break;
                } else if (result.get(j) == 0) cnt ++;
            }
        }

        for (int i = 0; i < n; i++) {
            if (i != n-1) System.out.print(result.get(i) + " ");
            else System.out.print(result.get(i));
        }

    }
}

File: 백준/Silver/1138. 한 줄로 서기/한 줄로 서기.py
import sys
input = sys.stdin.readline

n = int(input())
li = list(map(int,input().split()))
result = [0]*n

for i in range(n):
    cnt = 0
    for j in range(n):
        if cnt == li[i] and result[j] == 0:
            result[j] = i+1
            break
        elif not result[j]: cnt += 1

print(*result)